__tests__/content-endpoints.test.mjs

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env node
+
+/**
+ * Content Endpoints Test Suite
+ *
+ * Tests content API endpoints to verify they return proper responses
+ * for various content types including documentation, redirects, and metadata.
+ *
+ * Usage: npm run test:content-endpoints
+ */
+
+import { exec } from 'child_process'
+import { promisify } from 'util'
+
+const execAsync = promisify(exec)
+
+// Try to detect which port is available (3000 first, then 3001)
+async function detectBaseUrl() {
+	if (process.env.TEST_BASE_URL) {
+		return process.env.TEST_BASE_URL
+	}
+
+	try {
+		// Try port 3000 first (standard Next.js dev port)
+		await execAsync(
+			'curl -s --connect-timeout 2 http://localhost:3000/api/content/terraform/version-metadata',
```

Comment:
We default to port `8080` in UDR, as we default to `3000` on dev-portal and want to make it easy to run them both together.

https://github.com/hashicorp/web-unified-docs/blob/main/.env#L1